core/src/main/java/org/kohsuke/stapler/ResponseImpl.java

```diff
@@ -305,7 +305,7 @@ public OutputStream getCompressedOutputStream(HttpServletRequest req) throws IOE
         // CompressionFilter not available, so do it on our own.
         // see CompressionFilter for why this is not desirable
         setHeader("Content-Encoding","gzip");
-        return recordOutput(new FilterServletOutputStream(new GZIPOutputStream(super.getOutputStream())));
+        return recordOutput(new FilterServletOutputStream(new GZIPOutputStream(super.getOutputStream()), super.getOutputStream()));
```

Comment:
this is pretty ugly - there is no need for stapler to do gzip at all - see #130

Reply:
currently on 2 uses outside of stapler that I found - I have PRs for both, both of which means this can be completely removed in #130